usecase/vehicle.go

```diff
@@ -0,0 +1,24 @@
+package usecase
+
+import (
+	"github.com/edgarduran/Go-Dispatch-Bootcamp/models"
+	"github.com/edgarduran/Go-Dispatch-Bootcamp/service"
+)
+
+func GetAllVehicles() ([]models.Vehicle, error) {
+	vehicles, err := service.GetAllVehicles()
+	if err != nil {
+		return nil, err
+	}
+
+	return vehicles, nil
+}
+
+func GetVehicleById(id int) ([]models.Vehicle, error) {
+	vehicle, err := service.GetVehicleById(id)
+	if err != nil {
+		return nil, err
+	}
+
+	return vehicle, nil
+}
```

Comment:
`GetVehicleByID` should return a single instance of `models.Vehicle`, not a whole array of them.

```suggestion
func GetVehicleByID(id int) (models.Vehicle, error) {
	vehicle, err := service.GetVehicleByID(id)
	if err != nil {
		return nil, err
	}

	return vehicle, nil
}
```